iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractDcConverterAdder.java

```diff
@@ -0,0 +1,213 @@
+        return self();
+    }
+
+    public T setConnectableBus2(String connectableBus2) {
+        this.connectableBus2 = connectableBus2;
+        return self();
+    }
+
+    protected Optional<TerminalExt> checkAndGetTerminal2() {
+        if (bus2 != null || node2 != null) {
+            return Optional.of(new TerminalBuilder(voltageLevel.getNetworkRef(), this, ThreeSides.TWO)
+                    .setNode(node2)
+                    .setBus(bus2)
+                    .setConnectableBus(connectableBus2)
+                    .build());
+        }
+        return Optional.empty();
+    }
+
+    public T setIdleLoss(double idleLoss) {
+        this.idleLoss = idleLoss;
+        return self();
+    }
+
+    public T setSwitchingLoss(double switchingLoss) {
+        this.switchingLoss = switchingLoss;
+        return self();
+    }
+
+    public T setResistiveLoss(double resistiveLoss) {
+        this.resistiveLoss = resistiveLoss;
+        return self();
+    }
+
+    public T setPccTerminal(Terminal pccTerminal) {
+        this.pccTerminal = (TerminalExt) pccTerminal;
+        return self();
+    }
+
+    public T setControlMode(DcConverter.ControlMode controlMode) {
+        this.controlMode = controlMode;
+        return self();
+    }
+
+    public T setTargetP(double targetP) {
+        this.targetP = targetP;
+        return self();
+    }
+
+    public T setTargetVdc(double targetVdc) {
+        this.targetVdc = targetVdc;
+        return self();
+    }
+
+    protected void preCheck() {
+        boolean twoAcTerminals = bus2 != null || node2 != null;
+        if (controlMode == null) {
+            throw new ValidationException(this, "controlMode is not set");
+        }
+        if (Double.isNaN(targetP)) {
+            throw new ValidationException(this, "targetP is not set");
+        }
+        if (Double.isNaN(targetVdc)) {
+            throw new ValidationException(this, "targetVdc is not set");
+        }
+        if (twoAcTerminals && pccTerminal == null) {
+            throw new ValidationException(this, "converter has two AC terminals and pccTerminal is not set");
+        }
+        if (pccTerminal != null) {
+            var c = pccTerminal.getConnectable();
+            if (twoAcTerminals && !(c instanceof Branch<?> || c instanceof ThreeWindingsTransformer)) {
+                throw new ValidationException(this, "converter has two AC terminals and pccTerminal is not a line or transformer terminal");
+            } else if (!twoAcTerminals && !(c instanceof Branch<?> || c instanceof ThreeWindingsTransformer || c instanceof DcConverter<?>)) {
+                throw new ValidationException(this, "pccTerminal is not a line or transformer or the converter terminal");
+            }
+            if (c.getParentNetwork() != voltageLevel.getParentNetwork()) {
+                throw new ValidationException(this, "pccTerminal is not in the same parent network as the voltage level");
+            }
+        }
+    }
```

Comment:
In an LCC monopole with a 12-pulse bridge, is it possible to have the pccTerminal on the AC busbar where the two transformers might be connected?

Reply:
ok